src/main/clientHandlers/JSONExtractor.java

```diff
@@ -0,0 +1,35 @@
+package clientHandlers;
+
+import org.json.simple.parser.JSONParser;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.ParseException;
+
+public class JSONExtractor implements MessageExtractorInterface{
+    @Override
+    public PeripheralAction handleMessage(String input){
+        JSONParser parser = new JSONParser();
+        String action = "";
+        long id = -1;
+        try {
+            // Convert input string to JSON object, look for action description
+            Object inObj = parser.parse(input);
+            JSONObject message = (JSONObject) inObj;
+            JSONObject actionDescription = (JSONObject) message.get("action");
+            if(actionDescription == null){
+
+                return new PeripheralAction("NOP",-1);
```

Comment:
I doubt that it is the best idea that the jsonExtractor decides what to do when it not an action or a json string. I personally think it would be better to thrown an exception.